src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -314,6 +340,23 @@ public IRecordingDescriptor startRecording(
                 .findFirst();
     }
 
+    public void notifyRecordingStopped(String recordingName, String targetId) {
+        notificationFactory
+                .createBuilder()
+                .metaCategory(STOP_NOTIFICATION_CATEGORY)
+                .metaType(HttpMimeType.JSON)
+                .message(Map.of("recording", recordingName, "target", targetId))
+                .build()
+                .send();
+    }
+
+    public void cancelScheduledNotificationIfExists(String stoppedRecordingName) {
+        if (scheduledStopNotifications.containsKey(stoppedRecordingName)) {
```

Comment:
Also, I think we might want to add some thread synchronization when we're accessing this map. I believe it's possible that a client, or multiple clients, can send HTTP API requests in very tight timing to start and immediately stop a recording, which could lead to the Vert.x worker thread that handles scheduling this notification to still be underway doing it at this point, then another worker thread stops/deletes the recording, and then the first thread continues to schedule the notification for that recording after it has already been stopped or deleted. It would be very hard to replicate the behaviour and it would be a minor bug, but I think it's still possible and worth handling.